File: c++/array/FindMinimumInRotatedArray.cpp
class Solution {
public:
    int findMin(vector<int> &num) {
        int low = 0;
        int high = int(num.size()) - 1;
        while (low < high - 1) {
            int mid = (low + high) / 2;
            if (num[mid] > num[high]) {
                low = mid;
            } else {
                high = mid;
            }
        }
        
        return std::min(num[low], num[high]);
    }
};

File: archive/python/array/FindMinimumInRotatedArray.py
class Solution:
    # @param num, a list of integer
    # @return an integer
    def findMin(self, num):
        low = 0
        high = len(num) - 1
        while low < high - 1:
            mid = (low + high) / 2
            if num[mid] > num[high]:
                low = mid
            else:
                high = mid
                
        return min(num[low], num[high])
